android/app/src/main/java/com/microsoft/codepush/react/CodePushTelemetryManager.java

```diff
@@ -23,71 +33,103 @@ public CodePushTelemetryManager(Context applicationContext, String codePushPrefe
 
     public WritableMap getBinaryUpdateReport(String appVersion) {
         String previousStatusReportIdentifier = this.getPreviousStatusReportIdentifier();
+        WritableNativeMap reportMap = null;
         if (previousStatusReportIdentifier == null) {
-            this.recordDeploymentStatusReported(appVersion);
-            WritableNativeMap reportMap = new WritableNativeMap();
-            reportMap.putString("appVersion", appVersion);
-            return reportMap;
+            this.clearRetryStatusReport();
+            reportMap = new WritableNativeMap();
+            reportMap.putString(APP_VERSION_KEY, appVersion);
         } else if (!previousStatusReportIdentifier.equals(appVersion)) {
-            this.recordDeploymentStatusReported(appVersion);
-            WritableNativeMap reportMap = new WritableNativeMap();
+            this.clearRetryStatusReport();
+            reportMap = new WritableNativeMap();
             if (this.isStatusReportIdentifierCodePushLabel(previousStatusReportIdentifier)) {
                 String previousDeploymentKey = this.getDeploymentKeyFromStatusReportIdentifier(previousStatusReportIdentifier);
                 String previousLabel = this.getVersionLabelFromStatusReportIdentifier(previousStatusReportIdentifier);
-                reportMap.putString("appVersion", appVersion);
-                reportMap.putString("previousDeploymentKey", previousDeploymentKey);
-                reportMap.putString("previousLabelOrAppVersion", previousLabel);
+                reportMap.putString(APP_VERSION_KEY, appVersion);
+                reportMap.putString(PREVIOUS_DEPLOYMENT_KEY_KEY, previousDeploymentKey);
+                reportMap.putString(PREVIOUS_LABEL_OR_APP_VERSION_KEY, previousLabel);
             } else {
                 // Previous status report was with a binary app version.
-                reportMap.putString("appVersion", appVersion);
-                reportMap.putString("previousLabelOrAppVersion", previousStatusReportIdentifier);
+                reportMap.putString(APP_VERSION_KEY, appVersion);
+                reportMap.putString(PREVIOUS_LABEL_OR_APP_VERSION_KEY, previousStatusReportIdentifier);
+            }
+        }
+
+        return reportMap;
+    }
+
+    public WritableMap getRetryStatusReport() {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CODE_PUSH_PREFERENCES, 0);
+        String retryStatusReportString = settings.getString(RETRY_DEPLOYMENT_REPORT_KEY, null);
+        if (retryStatusReportString != null) {
+            clearRetryStatusReport();
```

Comment:
If `codePush.sync()` and `tryReportStatus()` both get called during an app resume, it could happen.. Maybe it's a rare scenario but since it's just one line needed to protect against such scenarios, i thought it wouldn't be much harm doing it.